### ISSUE TITLE
Different framesCount between in the original video file and the key in the JSON

### ISSUE THREAD

**Issue by YAYAYru** [heartexlabs/label-studio#3315]
**Describe the bug**
I am a label studio beginner in the labeling setup of Video Object Tracking. I noticed different frames counts between the original video file and key `framesCount` in JSON why? They must be the same. Maybe BUG? Or do I need to set up something? 

**To Reproduce**
1. Go to http://localhost:8080/projects
2. Create project and name it
3. import video mp4
4. Chose Video -> Video Object Tracking
5. After adding label click `</>` and see the error(does not match) `"framesCount": 106`  see the screenshot

**Expected behavior**
`framesCount` must be the same in Label studio and in Supervise.ly . By the way, you can check any video file with `cv2` in python

**Screenshots**
![image](https://user-images.githubusercontent.com/48862700/204079054-128097fd-0758-4c15-bd97-de75f72ea440.png)

**Environment (please complete the following information):**
 - OS: Ubuntu desktop 20.04
 - Label Studio Version 1.6.0

**Additional context**
In Screenshots see different `framesCount` of one video file in Label studio and in Supervise.ly , I'm sure the correct `framesCount` is in Supervise.ly

**Comment by makseq** [comment]
@YAYAYru Did you set framerate attribute in the Video tag of the labeling config correctly?

**Comment by YAYAYru** [comment]
> @YAYAYru Did you set framerate attribute in the Video tag of the labeling config correctly?

Thanks I found the attribute: 
```xml
<Video name="video" value="$video" framerate="25.0"/>
``` 
But I want to set the original framerate of each video file how? Because I have folders of video files with different framerates

**Comment by makseq** [comment]
Unfortunately only constant is support at this moment. 

```
<Video name="video" value="$video" framerate="$fps"/>
```
I would expect this one works, however it seems like a bug there and it doesn't :-(

**Comment by YAYAYru** [comment]
> Unfortunately only constant is support at this moment.
> 
> ```
> <Video name="video" value="$video" framerate="$fps"/>
> ```
> 
> I would expect this one works, however it seems like a bug there and it doesn't :-(
Yes, I didn't get the code `"$fps" `. Thanks for the info, I hope you fix the bug in the future.

**Comment by MetaHG** [comment]
> Unfortunately only constant is support at this moment.
> 
> ```
> <Video name="video" value="$video" framerate="$fps"/>
> ```
> 
> I would expect this one works, however it seems like a bug there and it doesn't :-(

Hi @makseq,

I am also affected by this issue. 
My dataset contains multiple videos with several different framerates. At the moment, transcoding the videos to an identical framerate is not an option.

Two questions related to this issue:
- Is there any ETA for this bug fix? 
- Is there a separate ticket somewhere to track this specific issue?

**Comment by makseq** [comment]
@MetaHG No, there is no separate ticket. I will try to force fixing it with our team.

**Comment by MetaHG** [comment]
Hi @makseq,

I have an additional question regarding the parameter `fps` in the labeling interface.

It is not uncommon in media files to have variable frame rate to maximize file compression. Does the video player and more generally label-studio allows to proceed accurately with such files? 
- If yes, does that mean that `fps` needs to be sepcified as the average fps?
- If no, what would exactly be the issue with such files? Is it only a video player synchronization issue? The annotation format adopted by label-studio looks suitable even for media files with a variable framerate as label-studio keyframes (not video encoding keyframes) are all saved with their corresponding frame number.

**Comment by hlomzik** [comment]
Hi, @MetaHG!
Unfortunately browser doesn't have access to video frames, so that's very important to set `frameRate` for every video. Video is played well anyway, the only thing is affected is timeline, we detect current frame as (time passed / given frame rate), so it can be only linear.
So for dynamic frame rates current frame could be not exactly the correct one.

**Comment by MetaHG** [comment]
Hi,
Thank you for this information. This is what I suspected. Transcoding is necessary for such videos then. 

Thank you for your help :)

**Comment by hlomzik** [comment]
there is PR ready btw, you can try it: https://github.com/heartexlabs/label-studio/pull/3525

**Comment by makseq** [comment]
@MetaHG @YAYAYru PR was merged to develop, so you can find it in the latest versions of LS from the github develop branch and this one should work there:

```
<Video name="video" value="$video" framerate="$fps"/>
```

**Comment by yayayru** [comment]
@makseq The problem has not yet been resolved, I didn't get the code `"$fps"` . I have version Label Studio 1.7.2. :(

**Comment by makseq** [comment]
@YAYAYru are you sure you are on 1.7.2? How did find that it is not working? 

Just checked it, works as expected. 

```
<View>
   <Labels name="videoLabels" toName="video" allowEmpty="true">
     <Label value="Man" background="blue"/>
     <Label value="Woman" background="red"/>
     <Label value="Other" background="green"/>
   </Labels>
   
   <!-- Please specify FPS carefully, it will be used for all project videos -->
   <Video name="video" value="$video" framerate="$fps"/>
   <VideoRectangle name="box" toName="video" />
</View>

<!--{
 "video": "/static/samples/opossum_snow.mp4",
 "fps": 10
}-->
```

![image](https://user-images.githubusercontent.com/501780/230247280-8758295d-3c22-447e-b486-7d027857eae0.png)

`/version` page:
```
{
  "release": "1.7.2",
  "label-studio-os-package": {
    "version": "1.7.2",
    "short_version": "1.7",
    "latest_version_from_pypi": "1.7.2",
    "latest_version_upload_time": "2023-03-20T16:52:30",
    "current_version_is_outdated": false
  },

  "label-studio-os-backend": {
    "message": "fix: LSDV-4740: Video Rectangles are displaying while drawing (#1258)  ...",
    "commit": "3de2ace9b53cf3ab213054a9106079aaf61796a7",
    "date": "2023-03-20 18:10:51 +0400",
    "branch": "HEAD",
    "version": "1.7.2+0.g3de2ace"
  },

  "label-studio-frontend": {
    "message": "fix: LSDV-4740: Video Rectangles are displaying while drawing (#1258)",
    "commit": "9329687afa56d7491f96f4aa2df81b15c12adf7f",
    "branch": "ls-release/1.7.2",
    "date": "2023/03/20 11:03:14"
  },

  "dm2": {
    "message": "Trigger build",
    "commit": "2ee2ddaf9596b9e600bbaf231768acd16815e498",
    "branch": "ls-release/1.7.2",
    "date": "2023-03-09T16:36:40Z"
  },

  "label-studio-converter": {
    "version": "0.0.50"
  }
}
```

**Comment by yayayru** [comment]
Yes, I'm sure 1.7.2. I have default fps 24 for the video file. 
![image](https://user-images.githubusercontent.com/48862700/230294634-a676ac42-5876-48b3-bb1a-7ef634701afe.png)
![image](https://user-images.githubusercontent.com/48862700/230294676-cc4f1ffd-6dbe-4816-bddb-67ed4006d4b3.png)
The default frame rate is 24 fps, so 11348 frames correspond to $fps. The original video should have 23640 frames with a frame rate of 50 fps. See screenshot below.
![image](https://user-images.githubusercontent.com/48862700/230301817-772d0a1f-6b9d-47ed-92c4-b6c6d3522ac6.png)

**Comment by makseq** [comment]
Why don't you use the correct fps in the task json? 

```
{
 "video": "/static/samples/opossum_snow.mp4",
 "fps": 50
}
```
? 

or you can replace $ftp to 50 in the labeling config.

**Comment by ppskmg** [comment]
Не очевидная вещь со сдвигом кадров. В чём была проблема сразу сделать: загрузил файл программно подцепился рейт и такой же выдал при экспорте, кому надо пусть лезут правят рейт при необходимости, сделали головную боль на ровном месте. Зато суслик какой-то с frame rate 25.0...

Решение:

```python
 # Узнать фрейм рейт видео
  import cv2

  # Путь к видеофайлу
video_path = 'video_file.mp4' 

  # Открываем видео
  cap = cv2.VideoCapture(video_path)
  if not cap.isOpened():
      print(f"Error: Couldn't open video file {video_path}")
      exit()

  # Получение значения фреймрейта
  frame_rate = cap.get(cv2.CAP_PROP_FPS)
  print(f"Frame rate: {frame_rate} FPS")

  # Освобождаем ресурсы
  cap.release()
```

``` python
# перебить разметку под новый фреймрейт
import json

# Путь к файлу JSON с разметкой
json_path = 'project-...export_frames_file..json'

# Чтение файла JSON
with open(json_path, 'r') as file:
    data = json.load(file)

# Пересчет номеров кадров
def adjust_frame_number(frame_number, original_fps, target_fps):
    return int(frame_number * (target_fps / original_fps))

original_fps = 25.0
target_fps = 30.0

for item in data:
    for annotation in item['annotations']:
        for result in annotation['result']:
            for sequence_item in result['value']['sequence']:
                sequence_item['frame'] = adjust_frame_number(sequence_item['frame'], original_fps, target_fps)

# Сохранение обновленных данных в новый JSON файл
adjusted_json_path = 'adjusted_annotations.json'
with open(adjusted_json_path, 'w') as file:
    json.dump(data, file, indent=4)

print("JSON данные обновлены и сохранены в 'adjusted_annotations.json'")

```

**Comment by YoungriKIM** [comment]
I had the same issue, and adjusting the frame rate(25.0 >> 29.97002997002997) to match my video solved the problem. Thank you!

**Comment by makseq** [comment]
@ppskmg Label Studio avoids reading users data for security reasons. This occurs only in the browser, and browsers do not typically support obtaining the number of frames.

**Comment by makseq** [comment]
The fix was merged to the develop branch today.

**Comment by erklem** [comment]
Is this the fix you're referring to? 5093b7f98105a32c4bd22dfac8478de8d58bca47
